docs/library/util_wfifo/index.rst

```diff
@@ -0,0 +1,83 @@
+.. _util_wfifo:
+
+Util WFIFO
+===============================================================================
+
+.. hdl-component-diagram::
+
+The :git-hdl:`Util WFIFO <library/util_wfifo>` core
+is designed to downscale the clock rate of the RX data path.
+There are scenarios when the device clock (interface clock) is too high (above
+200 MHz), making it a challenge to integrate any processing cores between the
+device core and CPACK/DMA, because of the small timing margins. By reducing the
+clock rate of the data path, the user can easily integrate any custom processing
+core into the design.
+
+To define the correct configuration, the following questions need to
+be answered:
+
+#. What is the clock rate of the device core's data interface? (dout_clk)
+#. What is the data rate of the device core's data interface?
+   (``dout_valid@dout_clk``)
+#. What is the targeted clock rate of the data path (din_clk), and how we can
+   achieve it, respecting the main rule of thumb: **input data rate must be
+   equal to the output data rate**?
+
+If the device clock rate is equal to the device data rate, the only solution to
+downscale the clock rate is to increase the data width of the output ports of
+the FIFO. Currently the util_wfifo supports four data width ratios:
+1:1/1:2/1:4/1:8.
+
+Features
+--------------------------------------------------------------------------------
+
+* Supports Altera and Xilinx devices.
```

Comment:
This slipped somehow, and I was reminded of it when reviewing your PR with util_rfifo. Could you please add it?
```suggestion
* Supports Altera and AMD Xilinx devices.
```